app/scripts/modules/core/retry/retry.service.ts

```diff
@@ -0,0 +1,35 @@
+import {module} from 'angular';
+
+export class RetryService {
+
+  static get inject(): string[] {
+    return ['$timeout', '$q'];
+  }
+
+  constructor(private $timeout: ng.ITimeoutService, private $q: ng.IQService) {}
+
+  // interval is in milliseconds
+  public buildRetrySequence(func: Function,
+                            stopCondition: Function,
+                            limit: number,
+                            interval: number): ng.IPromise<any> {
+
+    const call: any = func();
+    const promise: ng.IPromise<any> = call.then ? call : this.$q.resolve(call);
+    if (limit === 0) {
+      return promise;
+    } else {
+      return promise.then((result: any) => {
+        if (stopCondition(result)) {
+          return result;
+        } else {
+          return this.$timeout(interval).then(() => this.buildRetrySequence(call, stopCondition, limit - 1, interval));
```

Comment:
Huh. The expects run and pass if you set the timeout interval to 0, though I'm not sure why. You could just do that, since I don't think it invalidates the tests.